core/shared/src/main/scala/zio/IO.scala

```diff
@@ -452,39 +452,39 @@ object IO {
     ZIO.lock(executor)(io)
 
   /**
-   *  @see [[zio.ZIO.mapN]]
+   *  @see mapN in [[zio.ZIO$ ZIO]]
```

Comment:
I wonder if we can just move the methods defined on `ZIOFunctions` to the `ZIO` companion object. As far as I can tell we're not getting any benefit from having the trait and it was both solve this problem and slightly simplify the code. I asked about this on Discord so we'll see if anyone else knows why we did it this way.

Reply:
Let me know, I can move the methods to the object and fix the links

Reply:
Yes, looking in the history it appears that this is a holdover from an older encoding where there were `LowerR` and `UpperE` type bounds that were abstract and then specified in objects extending this trait. So let's move everything to the companion object.

Reply:
Everything, or just `mapN` and `parMapN`?

Reply:
Everything and just delete the trait entirely.

Reply:
:+1:

Reply:
Done